plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -2965,9 +2966,16 @@ public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(C
             Table icebergTable = catalog.loadTable(session, table.getSchemaTableName());
 
             Set<Integer> partitionSpecIds = table.getSnapshotId().map(
-                            snapshot -> icebergTable.snapshot(snapshot).allManifests(icebergTable.io()).stream()
-                                    .map(ManifestFile::partitionSpecId)
-                                    .collect(toImmutableSet()))
+                            snapshot -> {
+                                try {
+                                    return icebergTable.snapshot(snapshot).allManifests(icebergTable.io()).stream()
+                                            .map(ManifestFile::partitionSpecId)
+                                            .collect(toImmutableSet());
+                                }
+                                catch (NotFoundException | UncheckedIOException e) {
```

Comment:
Could you explain how up picked up these exception classes in PR description?

Reply:
sure, updated. Please review.

Reply:
please add the stacktrace of the exceptions to showcase better what kind of issue you are after.